mllib/src/main/scala/org/apache/spark/ml/feature/LSH.scala

```diff
@@ -112,7 +113,8 @@ private[ml] abstract class LSHModel[T <: LSHModel[T]]
       numNearestNeighbors: Int,
       singleProbe: Boolean,
       distCol: String): Dataset[_] = {
-    require(numNearestNeighbors > 0, "The number of nearest neighbors cannot be less than 1")
+    require(numNearestNeighbors > 0 && numNearestNeighbors <= dataset.count(), "The number of" +
```

Comment:
This count() is expensive. Hm, I guess we have to compute it at least once, but, save it. Otherwise it is computed like 3 times per loop below.